corehq/form_processor/tests/utils.py

```diff
@@ -172,8 +172,36 @@ def _delete_all(db, viewname, **view_kwargs):
 )
 
 
+def partitioned(cls):
+    """
+    Marks a test to be run with the partitioned database settings in
+    addition to the non-partitioned database settings.
+    """
+    return attr(sql_backend=True)(cls)
+
+
+def only_run_with_non_partitioned_database(cls):
+    """
+    Only runs the test with the non-partitioned database settings.
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        return nottest(cls)
+
+    return cls
+
+
+def only_run_with_partitioned_database(cls):
```

Comment:
(i'm going to merge this so it gets in with today's deploy everywhere - let me know if there's anything you'd prefer i do about these test decorators and i can handle in a separate pr)